### PR TITLE
Added modules for use with codeception

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Added some modules that can be used for codeception testing.
- see https://github.com/SU-SWS/ace-gryphon/pull/2
- refactored tasks to improve maintainablility
- see https://github.com/SU-SWS/stanford_profile/pull/153

# Need Review By (Date)
- 3/27

# Urgency
- medium

# Steps to Test
1. review implementation in https://github.com/SU-SWS/ace-gryphon/pull/2

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by pookmish** [comment]
OK i think i'm done adding to this for codeception execution. this branch is currently being used on https://github.com/SU-HKKU/cardinal_service_profile/pull/1

File: src/CaravanTrait.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace StanfordCaravan;
+
+trait CaravanTrait {
+
+  public function tooldir(){
```
Comment (sherakama) [inline src/CaravanTrait.php]: This file needs linting. CC where you at?

File: config/composer.json [hunk 2]
```diff
@@ -12,24 +12,32 @@
...
     ],
     "require": {
+        "php": ">=7.1",
+        "behat/mink-selenium2-driver": "^1.3.1",
         "bex/behat-screenshot": "^1.2",
+        "cweagans/composer-patches": "^1.6",
         "dmore/behat-chrome-extension": "^1.0.0",
         "drupal/core": "^8.8.0",
         "drupal/drupal-extension": "^4.0",
+        "drush/drush": "^10.2",
```
Comment (sherakama) [inline config/composer.json:26]: Do we need drush 10? This might conflict with our stack repos as they need to be on Drush 9 at the moment.

Comment (pookmish) [reply config/composer.json:26]: so this composer.json is used on the test builds when building a drupal directory from scratch. its not a dependency of this repo.

File: src/CaravanTrait.php (hunk 1 above)
Comment (pookmish) [reply src/CaravanTrait.php]: there it is. i'll address those this afternoon.

**Comment by pookmish** [comment]
@sherakama that was alot of codeclimate issues. but its nice and fixed up now.

File: src/StanfordBehat/DrupalExtension/Context/SwsMinkContext.php [hunk 3]
```diff
@@ -283,37 +291,15 @@ protected function getAbsoluteFilePath($path) {
...
-
-  /**
-   * @when /^(?:|I )cancel the popup$/
-   */
-  public function cancelPopup() {
-    $this->getSession()->getDriver()->getWebDriverSession()->dismiss_alert();
-  }
-
-  /**
-   * @When /^(?:|I )should see "([^"]*)" in popup$/
```
Comment (sherakama) [inline src/StanfordBehat/DrupalExtension/Context/SwsMinkContext.php:300]: Why are these being removed?

**Comment by sherakama** [comment]
A lot has shifted in `src/Robo/Tasks/RoboFile.php` and I am not really well set up to test it other than in the PR you linked so I am going to make the assumption that it is working and if it breaks you will be all over it like maple syrup on pancakes.

**Comment by jbickar** [comment]
> all over it like maple syrup on pancakes

Like a hungry pooch on pickled Alpo

File: src/StanfordBehat/DrupalExtension/Context/SwsMinkContext.php (hunk 3 above)
Comment (pookmish) [reply src/StanfordBehat/DrupalExtension/Context/SwsMinkContext.php:300]: they never worked in the first place. you can't run javascript to test for an alert when the alert blocks javascript from running.

**Review by sherakama — APPROVED** [review]

File: src/StanfordBehat/DrupalExtension/Context/SwsMinkContext.php (hunk 3 above)
Comment (sherakama) [reply src/StanfordBehat/DrupalExtension/Context/SwsMinkContext.php:300]: Thanks for splain'n

**Pull request merged**